src/components/Navbar.jsx

```diff
@@ -0,0 +1,139 @@
+              </div>
+            </button>
+          ))}
+        </div>
+        <div className="flex gap-x-6">
+          {lngs && (
+            <div className="flex relative" ref={translationMenuRef}>
+              <button
+                onClick={() => {
+                  setIsTranslationMenuOpen(!isTranslationMenuOpen);
+                  setIsHamburgerMenuOpen(false);
+                }}
+                ref={translationMenuRef}
+              >
+                <BsTranslate size={24} />
+              </button>
+              {isTranslationMenuOpen && (
+                <div className="absolute -right-6 top-14 p-2 rounded-xl shadow-2xl bg-black/75">
+                  <ul className="w-fit">
+                    {lngs.map(({ FlagIcon, title, clickHandler }) => (
+                      <li
+                        className="flex gap-x-2 items-center py-1 px-2 text-sm rounded-lg hover:bg-gray-300/25"
+                        key={title}
+                        onClick={clickHandler}
+                      >
+                        <FlagIcon className="h-4" />
+                        <p>{title}</p>
+                      </li>
+                    ))}
+                  </ul>
+                </div>
+              )}
+            </div>
+          )}
+          <div className="flex relative lg:hidden" ref={hamburgerMenuRef}>
+            <button
+              onClick={() => {
+                setIsTranslationMenuOpen(false);
+                setIsHamburgerMenuOpen(!isHamburgerMenuOpen);
+              }}
+            >
+              <RxHamburgerMenu size={24} />
+            </button>
+            {isHamburgerMenuOpen && (
+              <div className="absolute -right-6 top-14 p-2 rounded-xl shadow-2xl bg-black/75">
+                <ul className="w-fit">
+                  {sections.map(({ title, clickHandler }) => (
+                    <li
+                      className="flex gap-x-2 items-center py-1 px-2 text-sm rounded-lg min-w-32 hover:bg-gray-300/25"
+                      key={title}
+                      onClick={clickHandler}
+                    >
+                      {title}
+                    </li>
+                  ))}
+                </ul>
+              </div>
+            )}
+          </div>
+          <a
+            className="hidden py-1 px-4 text-sm text-black bg-white rounded-full lg:block hover:bg-gray-100"
+            href="mailto:info@ariseasia.org"
+          >
+            Contact Us
+          </a>
+        </div>
+      </div>
+    </nav>
+  );
+};
+
+Navbar.propTypes = {
+  className: PropTypes.string,
+  sections: PropTypes.arrayOf(
+    PropTypes.shape({
+      title: PropTypes.string.isRequired,
+      clickHandler: PropTypes.func.isRequired,
+    }),
+  ).isRequired,
+  lngs: PropTypes.arrayOf(
+    PropTypes.shape({
+      FlagIcon: PropTypes.func.isRequired,
+      title: PropTypes.string.isRequired,
+      clickHandler: PropTypes.func.isRequired,
+    }),
```

Comment:
if the fields inside the `lngs` isRequired, then does `lngs` also need to be required? because the field `sections` above are also all required?

Reply:
`lngs` should be optional for sites with no translation available or ready. this means that the list of objects is optional but when provided, each object is required to have all those fields.